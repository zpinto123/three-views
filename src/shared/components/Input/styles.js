import { css } from "styled-components";
const styles = {
  input: css`
    padding-left: 10px !important;
    padding-right: 10px !important;
  `
};

export default styles;
