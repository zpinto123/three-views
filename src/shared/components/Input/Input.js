import React from "react";
import PropTypes from "prop-types";
import { TextField } from "@material-ui/core";
import { styleUtils } from "../../utils";
import styles from "./styles";

const { style } = styleUtils;
const SInput = style(TextField, styles.input);

const Input = ({ value, handleChange, margin }) => (
  <SInput value={value} onChange={handleChange} margin={margin} />
);

Input.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  handleChange: PropTypes.func.isRequired
};

Input.defaultProps = {
  value: "",
  margin: "normal"
};

export default Input;
