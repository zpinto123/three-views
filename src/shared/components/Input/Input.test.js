import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";

import Input from "./";

describe("<Input />", () => {
  it("Show Input correctly", () => {
    const mockCallBack = jest.fn();

    const component = renderer.create(
      <Input value="Hello, i'm an input!" handleChange={mockCallBack} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("onChange works correcly", () => {
    const mockCallBack = jest.fn();

    const component = mount(
      <Input value="Hello, i'm an input!" handleChange={mockCallBack} />
    );
    component.find("input").simulate("change");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
