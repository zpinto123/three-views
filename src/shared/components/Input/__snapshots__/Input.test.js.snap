// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Show Input correctly 1`] = `
.c0 {
  padding-left: 10px !important;
  padding-right: 10px !important;
}

<div
  className="MuiFormControl-root MuiTextField-root c0 MuiFormControl-marginNormal"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Hello, i'm an input!"
    />
  </div>
</div>
`;
